torch/lib/THD/base/tensors/generic/THTensor.cpp

```diff
@@ -209,8 +218,302 @@ auto THTensor<real>::free() -> THTensor& {
   return *this;
 }
 
-#define non_const_cast(tensor) const_cast<THTensor&>(dynamic_cast<const THTensor&>(tensor))
-#define non_const_long_cast(tensor) const_cast<THTensor<long>&>(dynamic_cast<const THTensor<long>&>(tensor))
+template<>
+auto THTensor<real>::diag(const Tensor& src, int k) -> THTensor& {
+  THTensor_(diag)(
```

Comment:
Sure, I meant everywhere where line wont exceed 80 characters. Sorry for not specifing that.